### PR TITLE
feat(2083): Add tests for weighted executor configuration

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

Screwdriver cluster has ability to support multiple build executors - k8s, k8s-vm, jenkins etc . Currently users can use annotation screwdriver.cd/executor to pin to a specific executor available in the build cluster, if they don't want to use the default executor selected by the cluster admin. But there is a need to support distributing builds to multiple executors with in a build cluster instead of always picking the default one.

## Objective

This PR implements a feature to select an executor based on the assigned weight configured in the cluster and keeps the default logic intact following this priority pattern.
executor annotations > executors with weight > default executor

## References

https://github.com/screwdriver-cd/screwdriver/issues/2083

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/jobs.js [hunk 1]
```diff
@@ -15,8 +39,11 @@ const executorPlugins = Object.keys(executorConfig).reduce((aggregator, keyName)
 
     return aggregator;
 }, []);
+
+const weightedExecutor = getWeightedExecutor(executorPlugins);
+
 const executor = new ExecutorRouter({
-    defaultPlugin: executorConfig.plugin,
+    defaultPlugin: weightedExecutor || executorConfig.plugin,
```
Comment (jithine) [inline lib/jobs.js:46]: this won't work. it's initialized only once. This has to happen randomly each time a build is processed.

Comment (pritamstyz4ever) [reply lib/jobs.js:46]: correct, I am making this change to run for every build.

Comment (jithine) [reply lib/jobs.js:46]: you should put the logic in `executor-router`

Comment (jithine) [reply lib/jobs.js:46]: not in this I believe.

Comment (pritamstyz4ever) [reply lib/jobs.js:46]: screwdriver-executor-router is included as dependency in API, but doesn't seem ti be used.
Other places are:
https://github.com/screwdriver-cd/queue-service/blob/master/plugins/worker/lib/jobs.js#L35
https://github.com/screwdriver-cd/buildcluster-queue-worker/blob/master/lib/jobs.js#L18

I have aded the changes to executor-router which should not impact existing logic. 
https://github.com/screwdriver-cd/executor-router/pull/23

**Pull request merged**